From c4524ab78d72ddaf9c8b5bfda3191f24005b713b Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Fri, 1 Apr 2022 11:20:29 +0300
Subject: [PATCH] Fix no-opengl build

Amend f6f8f6d003b19fe6460346a61f224c8edfc031e3 to fix
build when configured without opengl.

Fixes: QTBUG-102209
Change-Id: I8b88d5a697b01016976048df6811adbcfb8d6170
---
 src/runtimerender/qssgrendershadercache.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/runtimerender/qssgrendershadercache.cpp b/src/runtimerender/qssgrendershadercache.cpp
index fcd17f29..6bc6bc74 100644
--- a/src/runtimerender/qssgrendershadercache.cpp
+++ b/src/runtimerender/qssgrendershadercache.cpp
@@ -128,9 +128,11 @@ static void initBaker(QShaderBaker *baker, QRhi *rhi)
     case QRhi::OpenGLES2:
     {
         QSurfaceFormat format = QSurfaceFormat::defaultFormat();
+#if QT_CONFIG(opengl)
         auto h = static_cast<const QRhiGles2NativeHandles *>(rhi->nativeHandles());
         if (h && h->context)
             format = h->context->format();
+#endif
         if (format.profile() == QSurfaceFormat::CoreProfile && format.version() >= qMakePair(3, 3)) {
             outputs.append({ QShader::GlslShader, QShaderVersion(330) }); // OpenGL 3.3+
         } else {
